tasks/main.yml

```diff
@@ -187,3 +187,38 @@
     - not __realm_join_output.stderr is search("Already joined to this domain")
   changed_when: not __realm_join_output.stderr is
     search("Already joined to this domain")
+
+- name: Configure dynamic DNS updates
+  ini_file:
+    path: /etc/sssd/sssd.conf
+    state: present
+    section: "domain/{{ ad_integration_realm | lower }}"
+    option: "{{ item.key }}"
+    value: "{{ item.value }}"
+    create: true
+    owner: root
+    group: root
+    mode: 0600
+  loop:
+    - key: dyndns_update
+      value: "{{ ad_dyndns_update | string }}"
+    - key: dyndns_ttl
+      value: "{{ ad_dyndns_ttl | int }}"
+    - key: ad_dyndns_iface
+      value: "{{ ad_dyndns_iface | string if ad_dyndns_iface is not none }}"
```

Comment:
I think you need an `else` here
or - maybe you want to skip and omit the key if value is none?  Then you shouldn't do the "cast" here and instead do something like this:
```yaml
ini_file:
    path: /etc/sssd/sssd.conf
    state: present
    section: "domain/{{ ad_integration_realm | lower }}"
    option: "{{ item.key if item.value is not none else omit }}"
    value: "{{ item.value | string if item.value is not none else omit }}"
...
loop:
    - key: dyndns_update
      value: "{{ ad_dyndns_update }}"
    - key: dyndns_ttl
      value: "{{ ad_dyndns_ttl }}"
    - key: ad_dyndns_iface
      value: "{{ ad_dyndns_iface }}"
...
```
I'm not sure what `omit` will do in this context - if that doesn't work, then you'll have to dynamically construct the list for the `loop`

Reply:
There are two config options `dyndns_iface` and `dyndns_server` if not specifically configured their default values are determined by SSSD. I did notice I set the keys to the variable names, I will fix that shortly.

Regarding the need of the `else`. Is this just best practice or are we anticipating this breaking in the future? As it sits now the task correctly updates the `sssd.conf`.

Reply:
Without the else - what is the value that gets set in sssd.conf?  I know we have had issues in the past when a value was set that used an `if` but didn't have an `else` and an "indeterminate" value was used in the `else` case

Reply:
It is being set to `''` then completely excluded from the `sssd.conf` by the `when: item.value != ''`. But I agree indeterminate values are not a good situation. What if I add `else ''` to those two options?

Reply:
> It is being set to `''` then completely excluded from the `sssd.conf` by the `when: item.value != ''`. But I agree indeterminate values are not a good situation. What if I add `else ''` to those two options?

Works for me - thanks

Reply:
I just thought about the use case where if you had set either of these two options then decide to remove them, allowing SSSD determine the defaults, the current task will not remove the options as would happen with the `blockinfile` version. I am going to add a second task that will remove them if they are have a value of `none` or `''`.

I'll make these changes, test everything, then will push and update.

Reply:
Update has been pushed. Please give me your thoughts.

Also, I have a test playbook I have been using to test all of my changes. I haven't included it because I didn't want to mess up any automated testing it may interfere with. I would like to work on getting it included next.

Reply:
> Update has been pushed. Please give me your thoughts.
> 
> Also, I have a test playbook I have been using to test all of my changes. I haven't included it because I didn't want to mess up any automated testing it may interfere with. I would like to work on getting it included next.

The test playbooks go in the `tests/` directory and begin with `tests_`.  You can look at the existing ones for examples.  We don't have automated AD tests - our QE will set up an external AD server and run tests manually - take a look at https://github.com/linux-system-roles/ad_integration/blob/main/tests/tests_full_integration_dc.yml for an example of how to write a test that works if you have a server `ad` in your inventory, or is otherwise skipped if no `ad`.

Reply:
See also https://linux-system-roles.github.io/contribute.html "Running tests with tox-lsr and qemu" - all of the `tests/tests_*.yml` playbooks should run and pass using this method.